### PR TITLE
Support span and field name containing dot for `search_traces`

### PR DESCRIPTION
**Pull request by harupy**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/harupy/mlflow/pull/12172?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/12172/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 12172
```

</p>
</details>

### Related Issues/PRs

 #xxx

### What changes are proposed in this pull request?

`search_traces` with `extract_fields` doesn't work when span or field name contains dot(s). This PR updates the parsing logic to support span or field name contains dot(s).

### How is this PR tested?

- [x] Existing unit/integration tests
- [x] New unit/integration tests
- [x] Manual tests

### Does this PR require documentation update?

- [ ] No. You can skip the rest of this section.
- [x] Yes. I've updated:
  - [x] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

#### Should this PR be included in the next patch release?

`Yes` should be selected for bug fixes, documentation updates, and other small changes. `No` should be selected for new features and larger changes. If you're unsure about the release classification of this PR, leave this unchecked to let the maintainers decide.

<details>
<summary>What is a minor/patch release?</summary>

- Minor release: a release that increments the second part of the version number (e.g., 1.2.0 -> 1.3.0).
  Bug fixes, doc updates and new features usually go into minor releases.
- Patch release: a release that increments the third part of the version number (e.g., 1.2.0 -> 1.2.1).
  Bug fixes and doc updates usually go into patch releases.

</details>

- [ ] Yes (this PR will be cherry-picked and included in the next patch release)
- [x] No (this PR will be included in the next minor release)

### CODE REVIEW

**Comment by B-Step62** [comment]
@harupy lemme remove patch label as discussed offline

**Comment by harupy** [comment]
@B-Step62 thanks!

File: tests/tracing/test_utils.py [hunk 1]
```diff
@@ -54,6 +56,62 @@ def test_maybe_get_request_id():
             maybe_get_request_id(is_evaluate=True)
 
 
+@pytest.mark.parametrize(
+    ("field", "expected"),
+    [
+        # no dot
+        ("span.inputs", _ParsedField("span", "inputs", None)),
```
Comment (harupy) [inline tests/tracing/test_utils.py:63]: Backticks can appear in the current implementation so ``span`name.inputs`` is a valid extract field.

File: mlflow/tracing/utils.py [hunk 2]
```diff
@@ -257,6 +257,108 @@ def extract_span_inputs_outputs(
...
+
+    def parse_field_name(self):
+        if self.peek() == "`":
+            self.next()
+            field_name = self.consume_until_char_or_end("`")
+            if self.peek() != "`":
+                raise MlflowException.invalid_parameter_value(
+                    f"Expected closing backtick: {self.field}"
+                )
+            self.next()
```
Comment (B-Step62) [inline mlflow/tracing/utils.py:350]: Can we check if there is any chars left in the field?

**Review by B-Step62 — APPROVED** [review]
LGTM with few minor comments!🙌

File: mlflow/tracing/utils.py [hunk 3]
```diff
@@ -257,6 +257,108 @@ def extract_span_inputs_outputs(
     )
 
 
+class _PeekableIterator:
```
[the remark this answers is not in the document]
Comment (harupy) [reply mlflow/tracing/utils.py:260]: sounds good!

File: mlflow/tracing/utils.py [hunk 4]
```diff
@@ -257,6 +257,108 @@ def extract_span_inputs_outputs(
...
+        Consume characters until the specified character is encountered or the end of the
+        string. If char is None, consume until the end of the string.
+        """
+        consumed = ""
+        while (c := self.peek()) and c != char:
+            consumed += self.next()
+        return consumed
+
+    def parse_span_name(self):
+        if self.peek() == "`":
```
[the remark this answers is not in the document]
Comment (harupy) [reply mlflow/tracing/utils.py:313]: great idea!

File: mlflow/tracing/utils.py [hunk 5]
```diff
@@ -269,21 +371,8 @@ class _ParsedField(NamedTuple):
 
     @classmethod
     def from_string(cls, s: str) -> "_ParsedField":
```
[the remark this answers is not in the document]
Comment (harupy) [reply mlflow/tracing/utils.py:373]: let's remove this!

**Pull request merged**